commitizen/providers/scm_provider.py

```diff
@@ -79,6 +79,24 @@ def get_version(self) -> str:
         )
         if not matches:
             return "0.0.0"
+
+        branch_prerelease_map = self.config.settings.get("branch_prerelease_map")
+        current_branch = get_current_branch()
+
+        if branch_prerelease_map and current_branch in branch_prerelease_map:
+            release_type = branch_prerelease_map[current_branch]
+
+            if release_type:
+                prerelease_matches = [
+                    v for v in matches if v.is_prerelease and release_type in v.prerelease
+                ]
+                if prerelease_matches:
+                    return str(prerelease_matches[-1])
```

Comment:
due we want to only keep tags of the same prerelease type?   isn't it possible that I could be on `staging/rc` branch and the latest version is a beta version because no rc versions have been created from that beta version?

would it work to say that we want to filter prereleases that are "higher" than the prerelease that corresponds to our branch?

- alpha would filter, beta and rc.
- beta would filter rc. 
- rc would not filter any prereleases

or to put this in the reverse:

- rc would keep rc, beta, and alpha versions
- beta would keep beta, and alpha versions
- alpha would keep only alpha